src/Cli/dotnet/commands/dotnet-workload/install/WorkloadManifestUpdater.cs

```diff
@@ -271,5 +271,24 @@ private ManifestVersion GetInstalledManifestVersion(ManifestId manifestId)
 
         internal static PackageId GetManifestPackageId(SdkFeatureBand featureBand, ManifestId manifestId) =>
             new PackageId($"{manifestId}.Manifest-{featureBand}");
+
+        // Note: cannot use Directory.Move because it errors when copying across mounts
+        internal static void CopyDirectory(string sourcePath, string destPath)
```

Comment:
It looks like this changes behavior from moving to copying.  It's also no longer atomic, which I think was the reason for extracting to a temp folder and then moving in the first place.

So I think we should use `Directory.Move` when it's possible.  When it's not possible, we probably want to do something more resilient / atomic, such as copying it to the metadata folder (which we can probably assume is on the same volume) and then moving it from there.

I'm not sure if there's a way to detect up-front whether Directory.Move will work or whether we have to catch the failure.

cc @wli3

Reply:
I agree on using Directory.Move when possible, I'll make that change. However I don't think we can always depend on the metadata dir being in the same mount- for example, when we're copying advertising manifests into the user home I believe that's on a different mount on linux by default.

Reply:
Is it possible that one of the target folders can be a symlink that crosses mount points?

Reply:
we kind of giving up the atomic move when we decided on using OS temp dir. So this is expected.

Although I am surprised Directory.Move will just fail